src/htmx.js

```diff
@@ -2193,6 +2193,7 @@ return (function () {
                     historyCache.shift(); // shrink the cache and retry
                 }
             }
+            if (historyCache.length === 0) localStorage.removeItem("htmx-history-cache");
```

Comment:
can you add a comment here?

Also maybe this should be up near the top and avoid all the logic in this method?

Reply:
@1cg I added a comment and moved it to the top to exit early. Note however that this changes the current behavior w.r.t. the `htmx:historyItemCreated` that was triggered up until now. However, I'd argue that the new behavior is more correct.

Reply:
agree